packages/loader/container-loader/src/loader.ts

```diff
@@ -299,6 +299,13 @@ export class Loader extends EventEmitter implements ILoader {
         });
     }
 
+    public async resolveWithPendingOps(request: IRequest, pendingOps): Promise<Container> {
```

Comment:
resolveWithLocallySavedState?
What kind of protection do we have to ensure that such pending state is for this URI, and host is not making mistake of using it incorrectly? If possible, we should have resolve URI as part of pending state and compare it against URI we are working with